exporter/exporter.py

```diff
@@ -0,0 +1,182 @@
+def check_capella_version():
+    if is_capella_5_x_x():
+        raise RuntimeError(
+            f"{ERROR_PREFIX} - Team for Capella 5.x.x is not supported"
+        )
+
+
+def checkout_git_repository() -> pathlib.Path:
+    git_dir = pathlib.Path("/tmp/git")
+    git_dir.mkdir(exist_ok=True)
+
+    git_repo_url: str = os.environ["GIT_REPO_URL"]
+    git_repo_branch: str = os.environ["GIT_REPO_BRANCH"]
+
+    log.debug("Cloning git repository...")
+    subprocess.run(
+        ["git", "clone", git_repo_url, "/tmp/git"],
+        check=True,
+        env={
+            "GIT_USERNAME": os.getenv("GIT_USERNAME", ""),
+            "GIT_PASSWORD": os.getenv("GIT_PASSWORD", ""),
+            "GIT_ASKPASS": "/etc/git_askpass.py",
+        },
+    )
+    try:
+        subprocess.run(
+            ["git", "switch", git_repo_branch], check=True, cwd=git_dir
+        )
+    except subprocess.CalledProcessError:
+        log.warning(
+            "Git branch %s does not exist in repository %s",
+            git_repo_branch,
+            git_repo_url,
+        )
+        raise RuntimeError(
+            f"{ERROR_PREFIX} - Branch {git_repo_branch} does not exist on repository {git_repo_url}"
+        )
+
+    log.debug("Clone of git repository finished")
+    return git_dir
+
+
+def determine_model_dir(git_dir: pathlib.Path) -> pathlib.Path:
+    model_dir: pathlib.Path = git_dir
+    if GIT_ENTRYPONT:
+        model_dir = pathlib.Path(
+            "/tmp/git", str(pathlib.Path(GIT_ENTRYPONT).parent).lstrip("/")
+        )
+    return model_dir
+
+
+def check_dir_for_aird_file(path: pathlib.Path):
+    aird_files = list(path.glob("*.aird"))
+
+    if not len(aird_files) == 1:
+        raise RuntimeError(
+            f"{ERROR_PREFIX} - Entrypoint (if provided) or root directoy does not contain a .aird file"
+        )
+
+    if GIT_ENTRYPONT and (
+        not pathlib.Path(GIT_ENTRYPONT).name == aird_files[0].name
+    ):
+        raise RuntimeError(
+            f"{ERROR_PREFIX} - .aird file found in git entrypoint directory does not match git entrypoint aird"
+        )
+
+
+# T4C determines the project name based on the directory name
+def get_model_dir_with_project_name(path: pathlib.Path) -> pathlib.Path:
+    project_dir: pathlib.Path = pathlib.Path(f"/tmp/{T4C_PROJECT_NAME}")
+    project_dir.mkdir(parents=True)
+
+    for file in path.glob("*"):
+        shutil.copy2(file, project_dir)
+
+    return project_dir
+
+
+def run_exporter_script(model_dir: pathlib.Path) -> None:
+    t4c_repo_name = os.environ["T4C_REPO_NAME"]
+
+    command: list[str] = [
+        "/opt/capella/capella",
+        "-consoleLog",
+        "-data",
+        "workspace",
+        "-application",
+        "com.thalesgroup.mde.melody.collab.exporter",
+        "-closeserverOnFailure",
+        "false",
+        "-overrideExistingProject",
+        "true",
+        "-mergeDifferenceOnExistingProjects",
+        "true",
+        "-hostname",
+        os.environ["T4C_REPO_HOST"],
```

Comment:
Yes, I agree that `T4C_REPO_HOST` is not the best name here, but I used it because we also use it in our `backups` image (i.e., also in the backend and I assume we use it in some of our pipelines as well. So I would rather leave it as it is, or rename all usages of `T4C_SERVER_HOSTNAME` (so we don`t have two terms for the same thing).